test/unittests/floating_point_utils_test.cpp

```diff
@@ -0,0 +1,162 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <test/utils/floating_point_utils.hpp>
+
+using namespace fizzy::test;
+
+TEST(floating_point_utils, double_as_uint)
+{
+    EXPECT_EQ(FP(0.0).as_uint(), 0x0000000000000000);
+    EXPECT_EQ(FP(-0.0).as_uint(), 0x8000000000000000);
+    EXPECT_EQ(FP(FP64::Limits::infinity()).as_uint(), 0x7FF'0000000000000);
+    EXPECT_EQ(FP(-FP64::Limits::infinity()).as_uint(), 0xFFF'0000000000000);
+    EXPECT_EQ(FP(FP64::Limits::max()).as_uint(), 0x7FE'FFFFFFFFFFFFF);
+    EXPECT_EQ(FP(-FP64::Limits::max()).as_uint(), 0xFFE'FFFFFFFFFFFFF);
+    EXPECT_EQ(FP(FP64::Limits::min()).as_uint(), 0x001'0000000000000);
+    EXPECT_EQ(FP(-FP64::Limits::min()).as_uint(), 0x801'0000000000000);
+    EXPECT_EQ(FP(FP64::Limits::denorm_min()).as_uint(), 0x000'0000000000001);
+    EXPECT_EQ(FP(-FP64::Limits::denorm_min()).as_uint(), 0x800'0000000000001);
+    EXPECT_EQ(FP(1.0).as_uint(), 0x3FF'0000000000000);
+    EXPECT_EQ(FP(-1.0).as_uint(), 0xBFF'0000000000000);
+    EXPECT_EQ(FP(std::nextafter(1.0, 0.0)).as_uint(), 0x3FE'FFFFFFFFFFFFF);
```

Comment:
Damn these helpers, so this is the `1.0 - <smallest representable value>`

Reply:
Not exactly. This is the next representable value after `1.0` towards `0.0`. But should be something like `0.9999...`.